ethcore/verification/src/test_helpers.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Verification test helpers.
+
+use std::collections::HashMap;
+
+use blockchain::{BlockProvider, BlockChain, BlockDetails, TransactionAddress, BlockReceipts};
+use common_types::{
+	BlockNumber,
+	encoded,
+	verification::Unverified,
+	log_entry::{LogEntry, LocalizedLogEntry},
+};
+use ethereum_types::{BloomRef, H256};
+use parity_bytes::Bytes;
+
+pub struct TestBlockChain {
```

Comment:
This probably just moved but can you `derive(Default)` and remove the `default impl`?

Also, I think the `new` impl should be removed because it is the same the default but if is used a lot don't bother

Reply:
https://rust-lang-nursery.github.io/api-guidelines/interoperability.html#c-common-traits says "Note that it is common and expected for types to implement both Default and an empty new constructor. new is the constructor convention in Rust, and users expect it to exist, so if it is reasonable for the basic constructor to take no arguments, then it should, even if it is functionally identical to default." – which makes sense to me so I think I'll keep `new`.

Reply:
ok, then I learned something new. I thought default was for `"default"` and new with some params...

Reply:
Yes, me too. I went looking at the guidelines for the comment about `Clone` and saw this.